lib/ansible/modules/network/ios/ios_user.py

```diff
@@ -234,20 +233,22 @@ def add(command, want, x):
 
     def add_ssh(command, want, x=None):
         command.append('ip ssh pubkey-chain')
-        command.append(' no username %s' % want['name'])
         if x:
-            command.append(' username %s' % want['name'])
-            command.append('  key-hash %s' % x)
-            command.append('  exit')
-        command.append(' exit')
+            command.append('username %s' % want['name'])
+            command.append('key-hash %s' % x)
+            command.append('exit')
+        else:
+            command.append('no username %s' % want['name'])
+        command.append('exit')
 
     for update in updates:
         want, have = update
 
         if want['state'] == 'absent':
-            commands.append(user_del_cmd(want['name']))
-            add_ssh(commands, want)
-            continue
+            if have['sshkey']:
+                add_ssh(commands, want)
+            else:
+                commands.append(user_del_cmd(want['name']))
```

Comment:
This line is in `else` condition. 
If this user have sshkey in settings, it will remove user's sshkey at first time, and remove user at second time?